keylime/payloads/keylime-controller/local_action_node_failover.py

```diff
@@ -1,27 +1,23 @@
 import os
 import asyncio
+import subprocess
 
-async def execute(revocation):
-    """Node fail over script
+"""Node fail over script
 
-    This script will be sent to the keylime_agent running on a machine
-    which has an OpenShift client (oc) installed and configured.
+This script will be sent to the keylime_agent running on a machine
+which has an OpenShift client (oc) installed and configured.
 
-    When a node is compromised this script will run `oc adm cordon` and
-    `oc adm drain` commands to remove that worker node from the OpenShift
-    cluster
-    """
-    if revocation['type']!='revocation':
-        return
+When a node is compromised this script will run `oc adm cordon` and
+`oc adm drain` commands to remove that worker node from the OpenShift
+cluster
+"""
 
-    # TODO - get the node name
-    node_name = "XXX"
-    rc = os.system("oc adm cordon {}".format(node_name))
-    if rc != 0:
-        print("Could not cordon node {}: Command failed with code {}".format(node_name, rc))
+async def execute(revocation):
+    if revocation['type']!='revocation':
         return
 
-    rc = os.system("oc adm drain {}".format(node_name))
-    if rc != 0:
-        print("Could not drain node {}: Command failed with code {}".format(node_name, rc))
-        return
+    node_name = 'mpeters-okd-zxrh5-' + revocation['agent_id']
+    cmd = "KUBECONFIG=/etc/kubeconfig /usr/local/bin/oc adm cordon {}".format(node_name)
+    subprocess.call(cmd, shell=True)
+    cmd = "KUBECONFIG=/etc/kubeconfig /usr/local/bin/oc adm drain {} --force=true --ignore-daemonsets --delete-local-data".format(node_name)
```

Comment:
How come `--ignore-daemonsets` is used? Is there a critical component like the machine config daemonset that will be needed to `uncordon` the node later?

Reply:
Drain won't actually work if there are daemonsets and `--ignore-daemonsets` isn't set. And in the case of OC nodes there are likely to be daemonsets. Does have the added side-effect of leaving the node as close as possible to what it was when the compromise was noticed, which may or may not be a benefit (those containers could still do things depending on what they are).